src/status_im/ui/screens/desktop/main/chat/views.cljs

```diff
@@ -101,21 +103,56 @@
                   :number-of-lines 5}
       text]]))
 
-(views/defview message-with-timestamp [text {:keys [message-id timestamp outgoing content current-public-key] :as message} style]
+(def regx-url #"(?i)(?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9\-]+[.][a-z]{1,4}/?)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:\'\".,<>?«»“”‘’]){0,}")
+
+(def regx-tag #"#[a-z0-9\-]+")
+
+(defn put-links-in-vector [text]
+  (map #(map (fn [token]
+               (cond
+                 (re-matches regx-tag token) [:tag token]
+                 (re-matches regx-url token)  [:link token]
+                 :default (str token " ")))
+             (string/split % #" "))
+       (string/split text #"\n")))
+
+(defn link-button [[link-tag link]]
+  [react/touchable-highlight {:style {}
+                              :on-press #(case link-tag
+                                           :link (.openURL react/linking (http/normalize-url link))
+                                           :tag (re-frame/dispatch [:chat.ui/start-public-chat (subs link 1)]))}
+   [react/text {:style {:font-size 9
+                        :color colors/blue
+                        :padding-bottom 1}}
+    (str link " ")]])
+
+(views/defview message-with-timestamp
+  [text {:keys [message-id timestamp outgoing content current-public-key] :as message} style]
```

Comment:
This is of course fine as for implementing the feature, but in the future we should move this processing (detecting link by regex - on both desktop and mobile) to receive handling of the message, because calculating it (potentially) over and over in view can worsen the performance.

Reply:
True it's done that way in the mobile version as well (though different much weirder algorithm, we should unify at some point and remove from the view